src/components/ProductListing/ProductListing.jsx

```diff
@@ -27,7 +27,7 @@ const ProductListing = () => {
                   className="card-image b-radius-2 mt-2"
                 />
               </section>
-              { token ?  presentInArray(wishlistState.itemsInWishlist, _id) ? <RemoveFromWishlistBtn productId={_id} token={token}/> : <AddToWishlistBtn productData={{product: {_id, name, brand, category, discountPercent, imgURL, mrp, price, rating, type }}} token={token}/>  : <AddToWishlistBtnRedirect/>}
+              { token ?  presentInArray(wishlistState.itemsInWishlist, _id) ? <RemoveFromWishlistBtn productId={_id} token={token}/> : <AddToWishlistBtn productData={{product: {_id, name, brand, category, discount, discountPercent, imgURL, mrp, price, rating, type }}} token={token}/>  : <AddToWishlistBtnRedirect/>}
```

Comment:
My suggestion is to do the destructuring in the child component. Why? Doing the restructuring while rendering the component affects readability